src/posts/xarray-biology/index.md

```diff
@@ -0,0 +1,157 @@
+- The `scverse` family of software uses `anndata`, which, like `Xarray` provides labeled dimensions. There are plans to have `anndata` be powered by `anndata`
+- The `bioio` package, capable of reading a zoo of microscope formats, provides a function to get image stacks as a `DataArray`.
+- The `Xarray.DataTree` provides a natural representation of multiscale images and can interoperate well with the OME-NGFF model. The [multiscale-spatial-image](https://github.com/spatial-image/multiscale-spatial-image?tab=readme-ov-file#multiscale-spatial-image) project uses `DataTree` for precisely this purpose.
+
+## What has limited adoption by Biologists?
+
+Given the benefits of switching to `Xarray`, why aren't more biologists using it? Is it secretly not as good as this blog claims? I spent February and April 2025 reaching out to scientists and scientific software developers across various fields of biology to find out. From those discussions, I've found three core reasons have prevented biologists who are already using Python from adopting `Xarray`.
+
+### Awareness and Examples
+
+Many biologists have never heard of `Xarray`. This is because the geosciences community created Xarray, and there wasn't enough community overlap for the word to spread to biologists.
+
+A related issue is that the examples and tutorials are geoscience-centric, which was the source of my discouragement when I first read the Xarray docs.
+
+### Technical Barriers
+
+Once a potential user is convinced of `Xarray`'s value, they may still face technical barriers. Ranging from rough edges to missing features, however, none are insurmountable. These barriers can exist in `Xarray` or in other tools in their workflow not accepting `Xarray` as input.
+
+An example of a rough edge in `Xarray` is that, as of May 2025, you cannot use integers as keys in a `DataTree`. That is a problem, as integers are a natural key when tracking single-cell lineages. Rough edges like this one haven't been smoothed over yet because there has not been a large user base of biologists using `Xarray` and raising issues when they encounter problems.
+
+Until recently, a missing feature was the ability to have `Xarray` coordinates that are more flexible than a fully instantiated array. Allowing for analytical transforms and having more flexible coordinates is critical to supporting volumetric imaging applications. Excitingly, in February 2025, initial support for this was [merged](https://github.com/pydata/xarray/pull/9543)!
+
+Another area of potential improvement is support for sparse arrays. Many biology datasets are sparse, so robust support is a key feature. Xarray does currently support sparse arrays so if you have tried using them and ran into problems please [open an issue](https://github.com/pydata/xarray/issues/new/choose).
```

Comment:
From my perspective, this is a very important feature. Can you cite an important example data set including its memory footprint when represented by sparse and dense arrays? Perhaps something from the DNA / RNA sequencing domain?